apps/prairielearn/elements/pl-rich-text-editor/pl-rich-text-editor.js

```diff
@@ -13,6 +13,35 @@
     }
   });
 
+  class Counter {
+    constructor(unit, uuid, getText) {
+      this.unit = unit;
+      this.container = document.getElementById(`rte-counter-${uuid}`);
+      this.getText = getText;
+
+      // Eagerly populate the container
+      this.update();
+    }
+
+    calculate(text) {
+      if (this.unit === 'word') {
+        const trimmed = text.trim();
+        // Splitting empty text returns a non-empty array
+        return trimmed.length > 0 ? trimmed.split(/\s+/).length : 0;
+      } else if (this.unit === 'character') {
+        return [...text].length;
```

Comment:
Adding a comment here would clarify why we're not using `text.length` instead.
```suggestion
        // Use a spread so that Unicode characters are counted instead of utf-16 code units
        return [...text].length;
```